aiida_quantumespresso/parsers/base.py

```diff
@@ -3,15 +3,128 @@
 
 All `Parser` implementations in `aiida-quantumespresso` must use this base class, not `aiida.parsers.Parser`.
 """
-from aiida.parsers import Parser as _BaseParser
+import re
+import typing
+
+from aiida.parsers import Parser
+from aiida.common import AttributeDict
+from aiida.engine import ExitCode
+from aiida_quantumespresso.utils.mapping import get_logging_container
+from aiida_quantumespresso.parsers.parse_raw.base import convert_qe_time_to_sec
+
+__all__ = ('BaseParser',)
+
+
+class BaseParser(Parser):  # pylint: disable=abstract-method
+    """Custom ``Parser`` class for ``aiida-quantumespresso`` parser implementations."""
+
+    class_error_map = {}
+    class_warning_map = {}
+
+    base_error_map = {
+        'Maximum CPU time exceeded': 'ERROR_OUT_OF_WALLTIME',
+    }
+    base_warning_map = {
+        'Warning:': None,
+        'DEPRECATED:': None,
+    }
+
+    @classmethod
+    def get_error_map(cls):
+        """The full error map of the parser class."""
+        error_map = cls.base_error_map.copy()
+        error_map.update(cls.class_error_map)
+        return error_map
+
+    @classmethod
+    def get_warning_map(cls):
+        """The full error map of the parser class."""
+        warning_map = cls.base_warning_map.copy()
+        warning_map.update(cls.class_warning_map)
+        return warning_map
+
+    def _retrieve_parse_stdout(self, **kwargs) -> typing.Tuple[str, dict, AttributeDict]:
```

Comment:
Wouldn't call this `_retrieve` because it is a bit misleading with the usual meaning of retrieving in AiiDA. The files have actually already been retrieved, you are just reading them from the `retrieved` output node and parsing them. Think `parse_stdout` would be sufficient, but if you really want it more specific, maybe `parse_retrieved_stdout` would work?

Reply:
Edit, I see now that you have the `parse_stdout` classmethod already that does the pure parsing of the stdout content. Maybe this should be called `parse_stdout_from_retrieved` ?